shipyard_blueprints/sftp/shipyard_sftp/sftp.py

```diff
@@ -260,9 +261,16 @@ def get_sftp_client(self) -> paramiko.SFTPClient:
                 logger.debug(
                     f"Connecting to {self.host} using key-based authentication"
                 )
+
                 ssh = paramiko.SSHClient()
                 ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-                key = paramiko.RSAKey.from_private_key_file(self.key)
+
+                if self.key.startswith("-----BEGIN"):
```

Comment:
I had also considered:

os.path.isfile()
os.path.isdir()
os.path.exists()
regex match
try generating key from file and if that fails try as a string.